src/WebJobs.Script.WebHost/Extensions/FunctionMetadataExtensions.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Management.Models;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Extensions
+{
+    public static class FunctionMetadataExtensions
+    {
+        /// <summary>
+        /// Maps FunctionMetadata to FunctionMetadataResponse.
+        /// </summary>
+        /// <param name="functionMetadata">FunctionMetadata to be mapped.</param>
+        /// <param name="hostConfig">The host options</param>
+        /// <returns>Promise of a FunctionMetadataResponse</returns>
+        public static async Task<FunctionMetadataResponse> ToFunctionMetadataResponse(this FunctionMetadata functionMetadata, ScriptHostConfiguration hostConfig, string routePrefix)
```

Comment:
Mostly a copy from the v2 codebase. However, because the v1 runtime doesn't expose management interfaces (e.g. VFS), for the hrefs returned the below GetScmBaseUrl function is used.